File: backend/src/main/java/com/quickbase/repositories/CountryRepository.java
package com.quickbase.repositories;

import com.quickbase.entities.Country;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CountryRepository  extends JpaRepository<Country, Integer> {
	@Query(nativeQuery = true, value =
			"SELECT SUM(city.Population) FROM Country "
			+ "INNER JOIN State state ON Country.CountryId = state.CountryId "
			+ "INNER JOIN City city ON state.StateId = city.StateId "
			+ "GROUP BY Country.CountryId "
			+ "ORDER BY Country.CountryId")
	List<Integer> getEachCountryPopulation();
}

File: c#/src/QuickBase.Persistence/Repositories/CountryRepository.cs
using Microsoft.EntityFrameworkCore;
using QuickBase.Domain.Entities;
using QuickBase.Domain.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuickBase.Persistence.Repositories
{
    public class CountryRepository : ICountryDatabaseRepository
    {
        private readonly QuickBaseDbContext db;

        public CountryRepository(QuickBaseDbContext db)
        {
            this.db = db;
        }

        public async Task<List<Country>> GetCountries()
        {
            var countries = from country in db.Countries
                            join state in db.States on country.CountryId.GetValueOrDefault() equals state.CountryId
                            join city in db.Cities on state.StateId equals city.StateId
                            group new { country, city } by country.CountryName into countryGroup
                            select new Country()
                            {
                                Name = countryGroup.Key,
                                Population = countryGroup.Sum(x => x.city.Population.GetValueOrDefault())
                            };

            return await countries.ToListAsync();
        }
    }
}
